### PR TITLE
Fix business rules partly un deployment deployment state change issue

### PR DESCRIPTION
**Pull request by thushaaanthan**

### CODE REVIEW

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java [hunk 1]
```diff
@@ -403,9 +403,21 @@ private int getDeploymentStatus(String nodeURL, String siddhiAppName, int busine
...
         } catch (SiddhiAppsApiHelperException e) {
-            if (businessRuleStatus == TemplateManagerConstants.PARTIALLY_DEPLOYED) {
-                return TemplateManagerConstants.SIDDHI_APP_NOT_DEPLOYED;
+           if (businessRuleStatus == TemplateManagerConstants.PARTIALLY_DEPLOYED && e.getStatus() == 404) {
+               return TemplateManagerConstants.SIDDHI_APP_DEPLOYED;
+           } else  if (businessRuleStatus == TemplateManagerConstants.PARTIALLY_DEPLOYED) {
+               return TemplateManagerConstants.SIDDHI_APP_UNREACHABLE;
+           }
+           if (businessRuleStatus == TemplateManagerConstants.PARTIALLY_UNDEPLOYED && e.getStatus() == 404) {
+               return TemplateManagerConstants.SIDDHI_APP_NOT_DEPLOYED;
```
Comment (AnuGayan) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: Why this is SIDDHI_APP_NOT_DEPLOYED when there is a possibility of still deployed in the unreachable node(s).

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java [hunk 2]
```diff
@@ -403,9 +403,21 @@ private int getDeploymentStatus(String nodeURL, String siddhiAppName, int busine
...
+            } else if (businessRuleStatus == TemplateManagerConstants.PARTIALLY_UNDEPLOYED ) {
+               return TemplateManagerConstants.SIDDHI_APP_UNREACHABLE;
```
[the remark this answers is not in the document]
Comment (thushaaanthan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: changed

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java [hunk 3]
```diff
@@ -403,9 +403,21 @@ private int getDeploymentStatus(String nodeURL, String siddhiAppName, int busine
...
             }
+           if (businessRuleStatus == TemplateManagerConstants.DEPLOYED && e.getStatus() == 404) {
+               return TemplateManagerConstants.SIDDHI_APP_NOT_DEPLOYED;
+            } else if (businessRuleStatus == TemplateManagerConstants.DEPLOYED) {
+               return TemplateManagerConstants.SIDDHI_APP_UNREACHABLE;
```
[the remark this answers is not in the document]
Comment (thushaaanthan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: changed in the code

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java (hunk 1 above)
Comment (thushaaanthan) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: changed in the code

**Comment by minudika** [comment]
Currently, we are locally testing this feature and will merge this when it's done.

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java (hunk 1 above)
Comment (minudika) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: @thushaaanthan please put the commit in a comment for the fixes.

**Pull request merged**